scripts/xtensa-build-zephyr.py

```diff
@@ -447,6 +463,11 @@ def build_platforms():
 			signing_key = default_rimage_key
 		sign_cmd += ["--tool-data", str(rimage_config), "--", "-k", str(signing_key)]
 
+		global sof_version
+		if sof_version == "0.0.0":
```

Comment:
Can you move this test inside `get_sof_version()` ? Making the cache a private implementation detail.

```python
def get_sof_version(abs_build_dir):
    if sof_version:
          return sof_version

    with open(...

       sof_version = ...

    return sof_version
```

Reply:
Okay, moving this test inside get_sof_version() is better. Thanks for your suggestions @marc-hb I updated the PR to address all your review comments.